eng/pipelines/typespec-validation-all.yml

```diff
@@ -19,6 +19,23 @@ pr:
     - eng
     - specification/common-types
 
+resources:
+  pipelines:
+  - pipeline: typespec
+    source: typespec - CI
+    project: internal
+    trigger:
```

Comment:
Interesting I don't think we have used this before. I assume this is actually triggering this pipeline whenever one of those pipelines complete? Is that correct? How does it end up pulling the latest version of the artifacts from those pipelines?

fyi @benbp

Reply:
For main however though we are pinning specific versions so we wouldn't ever be testing with the dev packages in our main builds.

Reply:
The devops feature is documented here:

https://learn.microsoft.com/en-us/azure/devops/pipelines/process/pipeline-triggers?view=azure-devops

It triggers public pipeline "TypeSpec Validation - All" whenever internal pipelines "typespec - CI" or "typespec-azure - CI" finishes.

The one limitation of this feature, is it can only trigger a build on the "default" branch of the target pipeline.  So I needed to change the "default" branch of pipeline "TypeSpec Validation - All" from `main` to `typespec-next`.  I don't love this, but it works and the only downside I see, is users may need to select a different branch when triggering manually through the devops UI.

As Timothee said, we don't flow artifacts across these builds, we rely on publish/consume of the "next" tag through npmjs.

Reply:
Ok so the missing context is that this will end up trigger on the typespec-next branch and now I understand how this would work given that branch uses the "next" tag.